point_viewer_grpc/src/bin/octree_server.rs

```diff
@@ -64,6 +64,87 @@ impl proto_grpc::Octree for OctreeService {
         ctx.spawn(f)
     }
 
+    fn get_points_in_frustum(
+        &self,
+        ctx: RpcContext,
+        req: proto::GetPointsInFrustumRequest,
+        resp: ServerStreamingSink<proto::GetPointsInFrustumReply>,
+    ) {
+        use std::thread;
+
+        // This creates a async-aware (tx, rx) pair that can wake up the event loop when new data
+        // is piped through it.
+        let (tx, rx) = mpsc::channel(4);
+        let octree = self.octree.clone();
+        thread::spawn(move || {
+            // This is the secret sauce connecting an OS thread to a event-based receiver. Calling
+            // wait() on this turns the event aware, i.e. async 'tx' into a blocking 'tx' that will
+            // make this thread block when the event loop is not quick enough with piping out data.
+            let mut tx = tx.wait();
+
+            let projection_matrix = Matrix4::from(PerspectiveFov {
+                fovy: Rad::from(Deg((req.fov_y as f32).into())),
+                aspect: (req.aspect as f32).into(),
+                near: (req.z_near as f32).into(),
+                far: (req.z_far as f32).into(),
+            });
+            let req_view_position = req.view_position.clone().unwrap();
+            let req_view_direction = req.view_position.clone().unwrap();
+            let req_view_up = req.view_position.clone().unwrap();
+
+            let view_position = Point3::new(req_view_position.x, req_view_position.y, req_view_position.z);
+            // view_direction: unit vector defining the direction vector
+            let _view_direction = Vector3::new(req_view_direction.x, req_view_direction.y, req_view_direction.z);
+            // view_up: unit vector defining the up vector
+            let view_up = Vector3::new(req_view_up.x, req_view_up.y, req_view_up.z);
+            let view_center = view_position + view_up;
+
+            let view_transform: Matrix4<f32> = Transform::look_at(view_position, view_center, view_up);
```

Comment:
A comment on what that does would be appreciated. According to the cgmath docs, "Create a transformation that rotates a vector to look at center from eye, using up for orientation." – I'm not sure I understand what that means. What does this do with a vector?